stubs/netaddr/netaddr/ip/__init__.pyi

```diff
@@ -71,9 +71,9 @@ class IPAddress(BaseIP):
     def ipv4(self) -> Self: ...
     def ipv6(self, ipv4_compatible: bool = False) -> Self: ...
     def format(self, dialect: type[ipv6_verbose] | None = None) -> str: ...
-    def __or__(self, other: str | SupportsInt | SupportsIndex) -> Self: ...
-    def __and__(self, other: str | SupportsInt | SupportsIndex) -> Self: ...
-    def __xor__(self, other: str | SupportsInt | SupportsIndex) -> Self: ...
+    def __or__(self, other: ConvertibleToInt) -> Self: ...
+    def __and__(self, other: ConvertibleToInt) -> Self: ...
+    def __xor__(self, other: ConvertibleToInt) -> Self: ...
```

Comment:
I would actually be inclined to make these stricter, rather than less strict. The [docstrings](https://github.com/netaddr/netaddr/blob/e212d8e5e67b1f1e09b4fc7833331e320cea622f/netaddr/ip/__init__.py#L610-L635) say:

```
:param other: An `IPAddress` object (or other int-like object).
```

To me, a `str` doesn't count as an int-like object, and I can't see a great reason to pass a `str` in here.

```suggestion
    def __or__(self, other: SupportsInt | SupportsIndex) -> Self: ...
    def __and__(self, other: SupportsInt | SupportsIndex) -> Self: ...
    def __xor__(self, other: SupportsInt | SupportsIndex) -> Self: ...
```